public/app/features/alerting/unified/api/dataSourcesApi.ts

```diff
@@ -4,8 +4,12 @@ import { alertingApi } from './alertingApi';
 
 export const dataSourcesApi = alertingApi.injectEndpoints({
   endpoints: (build) => ({
-    getDataSourceSettings: build.query<Array<DataSourceSettings<DataSourceJsonData>>, void>({
+    getAllDataSourceSettings: build.query<Array<DataSourceSettings<DataSourceJsonData>>, void>({
       query: () => ({ url: 'api/datasources' }),
+      // we'll create individual cache entries for each datasource UID
+      providesTags: (result) => {
```

Comment:
who invalidates this tags?

Reply:
Not included in this PR, but they will be invalidated when we make changes to the data source. 

The new settings page will toggle the `wantsAlertsReceived` property of the data source to enable / disable it as a receiver.